### DISCUSSION TITLE
Select which status checks to apply for automerge in Azure Devops

### DISCUSSION THREAD

**Discussion by trisjedwards** [renovatebot/renovate#22173] [Suggest an Idea]
**What would you like Renovate to be able to do?**
I would like to be able to automerge using Renovate when certain status checks have passed in my PR, but not all. 
For example, our master branches have a required reviewer branch policy, which Renovate cannot fulfill on its own. 
The bot account is capable of ignoring/overriding all the status checks, but then we would be unable to merge only successful builds.

**Describe the solution you'd like**
I would like a configuration option for Azure Devops to select which status checks are required before renovate will automerge to the target branch.

**Describe alternatives you've considered**
I've considered just always automerging, or automerging into a branch with different policies and then manually approving the bulk changes into master.

**Additional context**
This is a request specifically for Azure Devops.

**Comment by rarkins** [comment]
This looks like it would need an implementation of the `requiredStatusChecks` config option for Azure DevOps specifically.

Things we need to determine:
- [ ] Are Azure DevOps status checks each identified by some type of unique string/ID that could be used within Renovate config? i.e. users would configure an array of checks to wait for
- [ ] Confirm that we can correlate this list of strings against the API result

**Comment by trisjedwards** [comment]
I'm pretty sure that the status checks are identified by Policies at the project level, with a PolicyType: https://docs.microsoft.com/en-us/rest/api/azure/devops/policy/types/list?view=azure-devops-rest-5.1#policytype 

And on a PR level, status checks are called Evaluations: https://docs.microsoft.com/en-us/rest/api/azure/devops/policy/evaluations/list?view=azure-devops-rest-5.1 which contains a reference to the policy type and status (Approved/Failed etc)

**Comment by JamieMagee** [comment]
This might be enabled as of #7398. I haven't had a chance to confirm with automerge, but branch statuses are set correctly.

**Comment by teamtitangithub** [comment]
Is it implemeneted? Please give an example in the dos
